Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -312,37 +336,23 @@ FieldValue DecodeFieldValueImpl(pb_istream_t* stream) {
   }
 }
 
-/**
- * Encodes a FieldValue *and* its length.
- *
- * When encoding a top level message, protobuf doesn't include the length (since
- * you can get that already from the length of the binary output.) But when
- * encoding a sub/nested message, you must include the length in the
- * serialization.
- *
- * Call this method when encoding a non top level FieldValue. Otherwise call
- * EncodeFieldValue[Impl].
- */
-void EncodeNestedFieldValue(pb_ostream_t* stream,
-                            const FieldValue& field_value) {
-  // Implementation note: This is roughly modeled on pb_encode_delimited (which
-  // is actually pb_encode_submessage), adjusted to account for the oneof in
-  // FieldValue.
-
+void Writer::EncodeNestedMessage(
+    const std::function<void(Writer&)>& encode_message_fn) {
```

Comment:
Google C++ style would have us use `Writer*` rather than `Writer&` as an argument to the callback. Mutable references should not be passed as arguments unless required by some existing API (e.g. `std::swap`).

Reply:
Done.

Our linter actually catches this on regular functions. Not on std::functions though.